src/components/SearchModal/ManageLists.tsx

```diff
@@ -0,0 +1,312 @@
+import { TokenList } from '@uniswap/token-lists'
+
+import useToggle from '../../hooks/useToggle'
+import { AppDispatch, AppState } from '../../state'
+import { acceptListUpdate, removeList, disableList, enableList } from '../../state/lists/actions'
+import { useIsListActive, useAllLists } from '../../state/lists/hooks'
+import { ExternalLink, LinkStyledButton, TYPE, IconWrapper } from '../../theme'
+import { parseENSAddress } from '../../utils/parseENSAddress'
+import uriToHttp from '../../utils/uriToHttp'
+import { ButtonEmpty, ButtonPrimary } from '../Button'
+
+import Column, { AutoColumn } from '../Column'
+import ListLogo from '../ListLogo'
+import Row, { RowFixed, RowBetween } from '../Row'
+import { PaddedColumn, SearchInput, Separator, SeparatorDark } from './styleds'
+import ListToggle from '../Toggle/ListToggle'
+import Card from '../Card'
+import { CurrencyModalView } from './CurrencySearchModal'
+import { UNSUPPORTED_LIST_URLS } from '../../constants/lists'
+import { useActiveWeb3React } from '../../hooks'
+
+const Wrapper = styled(Column)`
+  width: 100%;
+  height: 100%;
+`
+
+const UnpaddedLinkStyledButton = styled(LinkStyledButton)`
+  padding: 0;
+  font-size: 1rem;
+  opacity: ${({ disabled }) => (disabled ? '0.4' : '1')};
+`
+
+const PopoverContainer = styled.div<{ show: boolean }>`
+  z-index: 100;
+  visibility: ${props => (props.show ? 'visible' : 'hidden')};
+  opacity: ${props => (props.show ? 1 : 0)};
+  transition: visibility 150ms linear, opacity 150ms linear;
+  background-color: ${({ theme }) => theme.bg1And2};
+  border: 1px solid ${({ theme }) => theme.bg3};
+  box-shadow: 0px 0px 1px rgba(0, 0, 0, 0.01), 0px 4px 8px rgba(0, 0, 0, 0.04), 0px 16px 24px rgba(0, 0, 0, 0.04),
+    0px 24px 32px rgba(0, 0, 0, 0.01);
+  color: ${({ theme }) => theme.text2};
+  border-radius: 0.5rem;
+  padding: 1rem;
+  display: grid;
+  grid-template-rows: 1fr;
+  grid-gap: 8px;
+  font-size: 1rem;
+  text-align: left;
+`
+
+const StyledMenu = styled.div`
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  position: relative;
+  border: none;
+`
+
+const StyledTitleText = styled.div<{ active: boolean }>`
+  font-size: 16px;
+  overflow: hidden;
+  text-overflow: ellipsis;
+  font-weight: 600;
+  color: ${({ theme, active }) => (active ? theme.white : theme.text2)};
+`
+
+const StyledListUrlText = styled(TYPE.main)<{ active: boolean }>`
+  font-size: 12px;
+  color: ${({ theme, active }) => (active ? theme.white : theme.text2)};
+`
+
+const RowWrapper = styled(Row)<{ bgColor: string; active: boolean }>`
+  background-color: ${({ bgColor, active, theme }) => (active ? bgColor ?? 'transparent' : theme.bg2)};
+  transition: background-color 0.2s ease;
+  align-items: center;
+  padding: 16px;
+  border-radius: 20px;
+`
+
+function listUrlRowHTMLId(listUrl: string) {
+  return `list-row-${listUrl.replace(/\./g, '-')}`
+}
+
+function ListRow({ listUrl }: { listUrl: string }) {
```

Comment:
How about extracting the `ListRow` component into a separate file to make the code easier to read?

Reply:
The code is forked from Uniswap. I'd rather leave it as unchanged as possible in structure, in order to more easily apply any future improvements.